src/corehost/cli/deps_resolver.cpp

```diff
@@ -403,6 +417,75 @@ void deps_resolver_t::init_known_entry_path(const deps_entry_t& entry, const pal
     }
 }
 
+void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
+{
+    pal::string_t additional_deps_serialized = init.additional_deps_serialized;
+    pal::string_t fx_name = init.fx_name;
+    pal::string_t fx_ver = init.fx_ver;
+
+    if (additional_deps_serialized.empty())
+    {
+        return;
+    }
+
+    pal::string_t additional_deps_path;
+    pal::stringstream_t ss(additional_deps_serialized);
+
+    // Process the delimiter separated custom deps files.
+    while (std::getline(ss, additional_deps_path, PATH_SEPARATOR))
+    {
+        // If it's a single deps file, insert it in 'm_additional_deps_files'
+        if (ends_with(additional_deps_path, _X(".deps.json"), false))
+        {
+            if (pal::file_exists(additional_deps_path))
+            {
+                m_additional_deps_files.push_back(additional_deps_path);
+            }
+        }
+        // If it's not a single deps file, we'll assume it's a directory
+        // and process it only for portable apps since they carry
+        // FX name and version. 
+        // 
+        // Standalone apps do not carry that information but are published
+        // incontext of a FX/Ver/RID. Thus, explicit path to deps.json
+        // must be specified for them.
+        else if (m_portable)
+        {
+            // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+            append_path(&additional_deps_path, _X("shared"));
+            append_path(&additional_deps_path, fx_name.c_str());
+            append_path(&additional_deps_path, fx_ver.c_str());
```

Comment:
We will be looking up the version that app requested for (which is what `fx_ver` corresponds to). There are no roll-forwards in this picture since lightup does not have to co-relate to premise of rollforward (i.e. apply patches).

You are correct, though, that different apps can explicitly callout various patch versions and thus, Azure/Asp.net would need to have folders for all of them. Even if we were to use the roll forwarded version, Azure/ASP.NET would still need to list out all versions. 

@DamianEdwards @glennc @davidfowl What all versions are you planning to list out?